### DISCUSSION TITLE
Shadow clipped during animation due to stale bounding box at resting pose

### DISCUSSION THREAD

**Discussion by azhong-git** [google/filament#6991] [Q&A] [answered]
Hi Filament community, 

I am seeing clipped shadow when a gltf model is being animated in gltf-viewer. I think this is caused by culling from a stale bounding box based on resting pose (see the screenshot, the bounding box from resting pose is shown).

Could you help me confirm this is the case? If so, how to get over this? Does Filament support recomputing bounding box at a frame level? Or are there any other way of addressing this?

Thank you.

![Screenshot 2023-07-25 at 4 10 42 PM](https://github.com/google/filament/assets/36863876/6fc397d0-ffbf-4501-afab-47cf86e642ec)

**Comment by bejado** [comment, marked as answer]
Yes, the reason is likely due to an incorrect bounding box calculation. We do not take bone animations into account (only morph targets) when computing the bounding box, so you may need to expand it slightly so it captures the entire animation.
